cpp/benchmarks/common/generate_benchmark_input.cpp

```diff
@@ -44,7 +44,7 @@ auto deterministic_engine(unsigned seed) { return std::mt19937{seed}; }
  *  Computes the mean value for a distribution of given type and value bounds.
  */
 template <typename T>
-T get_distribution_mean(distribution_params<T> const& dist)
+auto get_distribution_mean(distribution_params<T> const& dist) -> T
```

Comment:
Wow, such style (auto type return first, then trailing return type later) is not always favored. I'm not a fan of it.

Reply:
It is a part of the `modernize-` changes.

Reply:
I turned this option off for RMM.  I think we should turn it off for cuDF too.

These are the disabled rules I added for RMM: 

```
-cppcoreguidelines-macro-usage,
-llvm-header-guard,
-modernize-use-trailing-return-type,
-readability-named-parameter
```

Reply:
This PR is only focusing on `modernize-` checks. What is the argument for turning off `-modernize-use-trailing-return-type`.

This:
```cpp
auto function_that_returns_string() -> std::string;
auto function_returning_an_int() -> int32_t;
auto function_with_long_return_type() -> std::unique_ptr<cudf::column>;
```
is much more readable than:
```cpp
std::string function_that_returns_string();
int32_t function_returning_an_int();
std::unique_ptr<cudf::column> function_with_long_return_type();
```

Reply:
This feels pretty subjective to me and may be worth punting to a follow-up after we can come to some consensus?

Reply:
Exactly what Vyas said. It is completely subjective which  of the above is more readable. The latter is more readable to me, for example. I think we discussed when I was enabling clang-tidy on RMM and I don't think I was alone in that opinion.  More importantly, we didn't want to enforce a rule that allows only one style when both styles are acceptable to the team.

Reply:
I'll roll the changes back.